cmd/spicedb/serve.go

```diff
@@ -315,16 +315,18 @@ func serveRun(cmd *cobra.Command, args []string) {
 		redispatch = remote.NewClusterDispatcher(v1.NewDispatchServiceClient(conn))
 	}
 
-	cachingRedispatch, err := caching.NewCachingDispatcher(redispatch, nil, "dispatch_client")
+	cachingRedispatch, err := caching.NewCachingDispatcher(nil, "local_caching")
 	if err != nil {
-		log.Fatal().Err(err).Msg("failed to initialize redispatcher cache")
+		log.Fatal().Err(err).Msg("failed to initialize dispatcher cache")
 	}
+	cachingRedispatch.SetDelegate(redispatch)
```

Comment:
I think I'm missing the importance of separating out `SetDelegate`

Reply:
We need to separate it from the constructor because the delegate needs this item to be set as `it's` delegate

Reply:
where is that happening?